### PR TITLE
Styling/detailing

### PR DESCRIPTION
**Pull request by monshan**
## What does this PR do (summary of changes)?

- Applies animation to the landing page and representative link on the Navigation component
- Replaces old image assets and adds icons that better align with the branding of the application
- Applies hover transitions and shadow on NavLinks
- Refactors some common styles into extends
- Media query is applied to SongCard component to achieve responsivity down to 360px width
- Moves App component into its own component directory and fixed its import paths
- Removes repetitive scss imports in component JS files

## How should it be tested?
 
- User can observe more visual cues that imply interactivity on links
- User can easily observe all relevant elements and reach them on all screen widths down to 360px (including mobile)

## Future Iterations:

 - Final styling PR
 - Future iterations of the application might have the Navigation component moved to the bottom of the page

### CODE REVIEW

File: src/components/SongCard/SongCard.scss [hunk 1]
```diff
@@ -50,3 +73,28 @@
...
+
+  .handle-song-btn {
+    right: 20px;
+    top: 4%;
+  }
+
+  .lyrics-btn {
+    top: 15%; 
+  }
+}
```
Comment (m1073496) [inline src/components/SongCard/SongCard.scss]: It looks like these media queries aren't applying on some views.

**Review by m1073496 — COMMENTED** [review]
This looks so good! Thank you for working on this! 
Other than the media queries, we also need to reapply the `margin-top` spacing on MySongLibrary view to keep this layout consistent with the SongLibrary view.

**Review by elizhahn — APPROVED** [review]

**Comment by m1073496** [comment]
[Styling -> media queries](https://trello.com/c/NAJ0ahWR/95-styling-media-queries)

**Comment by m1073496** [comment]
[Styling](https://trello.com/c/NCnkqOhP/96-styling)

**Pull request merged**